build-in-container/scripts/build.sh

```diff
@@ -0,0 +1,207 @@
+#!/bin/bash
+set -euo pipefail
+
+SOURCE_FOLDER="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/.."
+
+CBL_MARINER_GIT_URL="https://github.com/microsoft/CBL-Mariner.git"
+
+VERBOSE=1
+LOG_LEVEL=info
+CCACHE_DIR=$BUILD_OUT_BASE_DIR/ccache
+SPECS_DIR=$SOURCE_FOLDER/SPECS
+USE_CCACHE="y"
+ARTIFACT_PUBLISH_DIR=""
+LOG_PUBLISH_DIR=""
+ERRORS_OCCURRED=0
+
+readonly MARINER_RELEASE_TAG="2.0-stable"
+
+if [ -t 1 ]; then
+    CYAN="\e[36m"
+    RESET="\e[0m"
+else
+    CYAN=""
+    RESET=""
+fi
+
+function log() {
+    timestamp="$(date "+%F %R:%S")"
+    echo -e "${CYAN}+++ $timestamp $1${RESET}"
+}
+
+function cleanup() {
+    log "Cleaning up..."
+
+    if [[ -n $LOG_PUBLISH_DIR ]]; then
+        publish_build_logs
+    fi
+}
+
+function download_mariner_toolkit() {
+    if [ ! -d toolkit ]; then
+        if [ ! -d CBL-Mariner ]; then
+            log " -- Clone CBL-Mariner toolkit from github"
+            git clone \
+                --branch ${MARINER_RELEASE_TAG} \
+                --depth 1 \
+                ${CBL_MARINER_GIT_URL}
+        fi
+        log " -- Build CBL-Mariner toolkit"
+        sudo make \
+            -C CBL-Mariner/toolkit \
+            package-toolkit \
+            BUILD_DIR="$BUILD_DIR" \
+            CHROOT_DIR="$CHROOT_BASE_DIR" \
+            CONCURRENT_PACKAGE_BUILDS="$CHROOT_NB" \
+            CONFIG_FILE= \
+            LOG_LEVEL=info \
+            OUT_DIR="$OUT_DIR" \
+            REBUILD_TOOLS=y && \
+        rm -rf CBL-Mariner && \
+        tar -xzvf ${OUT_DIR}/toolkit-*.tar.gz
+    fi
+}
+
+# Build a list of specs in a spec folder with a list of remote repos
+# Expects toolchain and worker chroot to be present before being called.
+#
+# No arguments
+# Global variables expected to be defined: BUILD_DIR, CCACHE_DIR, CHROOT_BASE_DIR, CHROOT_NB, LOG_LEVEL, OUT_DIR, SPECS_DIR
+build_specs() {
+    sudo make -C toolkit build-packages \
```

Comment:
If we are going to oblige the user to use pre-baked commands we should at least pass `-j$(nproc)` or something simliar.